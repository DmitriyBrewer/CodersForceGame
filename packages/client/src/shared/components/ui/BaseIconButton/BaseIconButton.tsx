import { IconButton } from '@mui/material'

const BaseIconButton = IconButton

export default BaseIconButton
