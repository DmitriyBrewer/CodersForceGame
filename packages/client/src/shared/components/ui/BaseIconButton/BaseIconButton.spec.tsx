import { render, screen } from '@testing-library/react'

import '@testing-library/jest-dom'
import BaseIconButton from './BaseIconButton'

describe('<BaseIconButton />', () => {
  const handleClick = jest.fn()

  it('Snapshot is correct', () => {
    const { asFragment } = render(<BaseIconButton onClick={handleClick} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('Renders correctly with default props', () => {
    render(<BaseIconButton />)
    const button = screen.getByRole('button')

    expect(button).toMatchSnapshot()
  })
})
