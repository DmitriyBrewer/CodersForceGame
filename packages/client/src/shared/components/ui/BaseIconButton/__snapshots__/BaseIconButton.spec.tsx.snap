// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseIconButton /> Renders correctly with default props 1`] = `
<button
  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
  tabindex="0"
  type="button"
>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`<BaseIconButton /> Snapshot is correct 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
